tokio/src/runtime/task/core.rs

```diff
@@ -168,11 +168,14 @@ pub(crate) struct Header {
     /// The id is not unset when removed from a list because we want to be able
     /// to read the id without synchronization, even if it is concurrently being
     /// removed from the list.
-    pub(super) owner_id: UnsafeCell<Option<NonZeroU64>>,
+    pub(super) owner_id: UnsafeCell<Option<NonZeroU32>>,
 
     /// The tracing ID for this instrumented task.
     #[cfg(all(tokio_unstable, feature = "tracing"))]
     pub(super) tracing_id: Option<tracing::Id>,
+
+    /// The task's ID, is used for deciding which list to put it in.
+    pub(super) task_id: Id,
```

Comment:
We already store the task id. It's in the core struct.

Reply:
Does the `header_lte_cache_line` test still pass when you put it here?

Reply:
> header_lte_cache_line

Yes. I reduce the size of `owner_id` to u32, I believe this size for owner_id is big enough. And I noticed that the `#[repr(C)]` is used, current struct fields 


I noticed that the tag a is used

Reply:
We could consider removing the task_id from the Core struct and, when a task is selected into Harness for execution, pass the task_id in the header to the Core method by parameter.

I'm not sure if this is the best approach.

Reply:
Why don't we expand using the existing `owner_id` field in `Header` struct instead of moving the task_id from `Core` to `Header`?

What we should do is that owner_id is still 64 bits(in current version I change it to u32), but now its lower 32 bits are the index of lists, and the upper 32 bits are the unique ID of ownedTasks. Of cause, we can use another u32 field to store the index, and also use u32 to store owner_id instead of u64.

When we need to remove the task from OwnedTasks, we use the lower 32 bits of the `owner_id` to find the list store it, we don't do caculating for index every time, it already caculated.

It can decouple index implementation details(which depends on task_id) from `Header` struct. Header owns the index of list have the tasks, instead of having the data to calculate index.

@Darksonn Do you think my above thought is more reasonable than the current PR design?